video/out/drm_atomic.c

```diff
@@ -458,3 +485,93 @@ bool drm_mode_destroy_blob(int fd, struct drm_mode *mode)
 
     return (ret == 0);
 }
+
+static inline uint16_t clamp16(float x)
+{
+    return (uint16_t)MPCLAMP(x, 0.0, 65535.0);
+}
+
+static inline uint16_t clamp16_round(float x)
+{
+    return clamp16(roundf(x * 50000.0));
+}
+
+int drm_create_hdr_metadata(struct drm_atomic_context *ctx, struct drm_hdr *hdr_data)
+{
+#if HAVE_DRM_HDR
+    if (mp_colorspace_equal(hdr_data->new_csp, hdr_data->current_csp))
+        return 0;
+
+    if (hdr_data->blob_id != 0)
+        drm_destroy_hdr_metadata(ctx, hdr_data);
+
+    struct hdr_output_metadata metadata = {0};
+    metadata.metadata_type = DRM_STATIC_METADATA_TYPE1;
+    metadata.hdmi_metadata_type1.metadata_type = DRM_STATIC_METADATA_TYPE1;
+
+    hdr_data->current_csp = hdr_data->new_csp;
+    struct mp_colorspace *csp = &hdr_data->current_csp;
+    struct mp_csp_primaries colors = mp_get_csp_primaries(csp->primaries);
+
+    switch(csp->gamma) {
```

Comment:
Nit: the style here is off (space between switch and `(`) - might want to check the rest of the patch/series.